subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/AbstractMutableModuleComponentResolveMetadata.java

```diff
@@ -259,6 +261,18 @@ public ImmutableAttributesFactory getAttributesFactory() {
         return attributesFactory;
     }
 
+    @Override
+    public void belongsTo(ComponentIdentifier platform) {
```

Comment:
Yup. I think belonging to different platforms makes sense when there are "real" published platforms, because the versions can be different in this case (we trust the definition of the platform). For "virtual" platforms, it probably doesn't make sense.

Reply:
At this stage we need to either a) prohibit multiple 'belongsTo' b) make 'belongsTo' overwrite previous call or c) add test coverage for this (including a bunch of permutations).

I'd probably go for a) for now, and add an issue to allow multiples and add test coverage.
The problem with b) is that changing it later would be a (minor) breaking change.

Reply:
I think c) shouldn't be too hard to add. I'll give it a try.